src/dmd/dsymbolsem.d

```diff
@@ -2018,6 +2023,69 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
          */
         addEnumMembers(ed, sc, sc.getScopesym());
 
+        if (sc.flags & SCOPE.Cfile)
+        {
+            /* C11 6.7.2.2
+             */
+            ed.memtype = Type.tint32; // C11 6.7.2.2-4 implementation defined
+            int nextValue = 0;        // C11 6.7.2.2-3 first member value defaults to 0
+
+            void emSemantic(EnumMember em, ref int nextValue)
+            {
+                static void errorReturn(EnumMember em)
+                {
+                    em.errors = true;
+                    em.semanticRun = PASS.semanticdone;
+                }
+
+                em.semanticRun = PASS.semantic;
+                em.type = Type.tint32;
+                em.linkage = LINK.c;
+                em.storage_class |= STC.manifest;
+                if (em.value)
+                {
+                    Expression e = em.value;
+                    assert(e.dyncast() == DYNCAST.expression);
+                    e = e.expressionSemantic(sc);
+                    e = resolveProperties(sc, e);
+                    e = e.integralPromotions(sc);
+                    e = e.ctfeInterpret();
+                    if (e.op == TOK.error)
+                        return errorReturn(em);
+                    auto ie = e.isIntegerExp();
+                    if (!ie)
+                    {
+                        // C11 6.7.2.2-2
+                        em.error("enum member must be an integral constant expression, not `%s` of type `%s`", e.toChars(), e.type.toChars());
+                        return errorReturn(em);
+                    }
+                    const sinteger_t v = ie.toInteger();
+                    if (v < int.min || v > uint.max)
```

Comment:
Just a note, pedantically this would be `(v < int.min || v > int.max)`, but I think it's fine to allow `1 << 31` as an enum value for compatibility with glibc's `sys/epoll.h`.

Reply:
C11 says "representable as an int". I interpreted that to mean "fits in an int", and this more generous view allows people to create enum bit flags and use all 32 bits.